### PR TITLE
listenOnly

### PR DESCRIPTION
**Pull request by kiroxas**

### CODE REVIEW

**Review by gaetanchamparnaud — APPROVED** [review]

**Review by axelgaillardasmodee — CHANGES_REQUESTED** [review]
This function does nothing to ensure you'll be the only one to listen to a given event, nor, as the name may suggest, that you'll only listen to this particular event. All it does is changing your callback for a particular event, if you already have one.

I suggest you give it another name, like changeCallback, and rewrite the misleading comment

File: test/EventCollectorTests.cpp [hunk 1]
```diff
@@ -27,6 +27,60 @@ TEST(EventCollector, SimpleTest)
...
+        
+    bus.notify(SimpleEvent{ 3 });
+    EXPECT_EQ(callCount, 2);
+}
+
+TEST(EventCollector, SimpleLoopTest)
+{
+    struct SimpleEvent
+    {
+        int value;
```
Comment (ddway2) [inline test/EventCollectorTests.cpp:63]: default value ?

**Review by ddway2 — APPROVED** [review]

**Comment by kiroxas** [comment]
> This function does nothing to ensure you'll be the only one to listen to a given event, nor, as the name may suggest, that you'll only listen to this particular event. All it does is changing your callback for a particular event, if you already have one.
> 
> I suggest you give it another name, like changeCallback, and rewrite the misleading comment

@axelgaillardasmodee 
Change callback is not a good name too, as if you do not have one already it will not "change" anything. But I'm all ears for a good name

File: test/EventCollectorTests.cpp (hunk 1 above)
Comment (kiroxas) [reply test/EventCollectorTests.cpp:63]: we are in tests, nobody cares

**Review by marou-asmodee — APPROVED** [review]

File: test/EventCollectorTests.cpp [hunk 2]
```diff
@@ -27,6 +27,60 @@ TEST(EventCollector, SimpleTest)
...
+    };
+
+    Dexode::EventBus bus;
+    int callCount = 0;
+    {
+        Dexode::EventCollector listener{ &bus };
+        for (int i = 0; i < 5; ++i)
```
Comment (PaoloBaerlocherAsmodee) [inline test/EventCollectorTests.cpp:70]: You should define a constant for 5 (used here and at line 81).

**Review by PaoloBaerlocherAsmodee — APPROVED** [review]

**Comment by axelgaillardasmodee** [comment]
`setCallback` sounds good enough

**Pull request merged**